### ISSUE TITLE
[expo-av] iOS: Reloading JS on physical device causes audio recording error 

### ISSUE THREAD

**Issue by computerjazz** [expo/expo#8948]
### Summary of Issue (just a few sentences)
When initially launching an audio recording app on a physical iOS device, recording works as expected. However, if you reload JS (shake phone, select "Reload") and attempt to start a recording, there is a native error.

I have traced it at least partly to this line:
https://github.com/expo/expo/blob/81e7b871187f304b428e8acb75581ce34167730b/packages/expo-av/ios/EXAV/EXAV.m#L795

`prepareToRecord` returns false after reloading JS.

NOTE: Does not happen on simulator on on expo snack. Only on bare expo app running on a physical iPhone.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
Platform: iOS -- iPhone 6

```
  Expo CLI 3.21.5 environment info:
    System:
      OS: macOS 10.15.5
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.9.1 - ~/.nvm/versions/node/v12.9.1/bin/node
      Yarn: 1.19.1 - ~/.yarn/bin/yarn
      npm: 6.10.2 - ~/.nvm/versions/node/v12.9.1/bin/npm
      Watchman: 4.7.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 11.1/11A1027 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~37.0.3 => 37.0.12 
      react: ~16.9.0 => 16.9.0 
      react-dom: ~16.9.0 => 16.9.0 
      react-native: ~0.61.5 => 0.61.5 
      react-native-web: ~0.11.7 => 0.11.7 
    npmGlobalPackages:
      expo-cli: 3.21.5

```

### Reproducible Demo
https://github.com/computerjazz/expo-av-bug

### Steps to Reproduce
- Shake phone
- Select "Reload"
- Attempt to start recording
- View error in log:

```
EXPO AV: startRecording error Error: Prepare encountered an error: recorder not prepared.
    at Object.fn [as callMethod] (NativeModules.js:99)
    at Object.NativeModulesProxy.<computed>.<computed> [as prepareAudioRecorder] (NativeModulesProxy.js:15)
    at Recording.prepareToRecordAsync$ (Recording.js:235)
    at tryCatch (runtime.js:45)
    at Generator.invoke [as _invoke] (runtime.js:274)
    at Generator.prototype.<computed> [as next] (runtime.js:97)
    at tryCatch (runtime.js:45)
    at invoke (runtime.js:135)
    at runtime.js:170
    at tryCallTwo (core.js:45)
```

### Expected Behavior vs Actual Behavior

Expected: Recording begins without error after reloading JS
Actual: Recording errors out after reloading JS

**Comment by computerjazz** [comment]
I have traced the issue to this commented-out line, which prevents the audio session from re-adding the module on JS refresh:
https://github.com/expo/expo/blame/master/packages/expo-av/ios/EXAV/EXAV.m#L117

It looks like commenting out that line was included as a temporary workaround for a different issue -- can it be reverted?

**Comment by IjzerenHein** [comment]
Hi! I've created a fix for this in PR #9283.
Thanks for the elaborate issue report 👍

**Comment by IjzerenHein** [comment]
Hi! We've released a test-version with the fix for this issue.

You can try it out in a bare project using `yarn install expo-av@8.4.0`.
And you can find the changelog here: https://github.com/expo/expo/blob/master/packages/expo-av/CHANGELOG.md

Let us know whether this fixes the problem for you!
